src/libs/PusherConnectionManager.js

```diff
@@ -16,16 +17,20 @@ function init() {
     }));
 
     /**
-     * Events that happen on the pusher socket are used to determine if the app is online or offline.
-     * The offline setting is stored in Onyx so the rest of the app has access to it.
-     *
      * @params {string} eventName
      */
-    Pusher.registerSocketEventCallback((eventName, data) => {
+    Pusher.registerSocketEventCallback((eventName, error) => {
         switch (eventName) {
             case 'error':
-                Log.info('[PusherConnectionManager] error event', false, {error: data});
-                Session.reauthenticatePusher();
+                if (error && error.type === 'PusherError' && error.data.code === 1006) {
```

Comment:
NAB question, we are guaranteed to have the `data` field be present with `code` defined?

Reply:
(given an `error` event name)

Reply:
Ah good catch. I think it's probably unsafe to assume that.